xline/src/storage/auth_store/backend.rs

```diff
@@ -66,7 +66,8 @@ pub(crate) struct AuthStoreBackend {
     /// Key Index
     index: Index,
     /// DB to store key value
-    db: DB,
+    // db: DB<Mutex<HashMap<Vec<u8>, Vec<u8>>>>,
+    db: DB<Memory>,
```

Comment:
I think we shouldn't hardcode `DB<Memory>` here. 
First, We can add a generic parameter for `AuthStoreBackend`, like `AuthStoreBackend<S: Storage>` and let `db` be an object in type `S`. So we can inject the given storage backend which comes from a config file or command line arguments into XlineServer.
Second, We can define a `MemStore` in the `storage` module, and implement `Storage` trait for it.

Reply:
fixed